evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -1407,3 +1409,29 @@ def test_partial_dependence_datetime_extra(
         assert len(part_dep["partial_dependence"]) == 10
         assert len(part_dep["feature_values"]) == 10
     assert not part_dep.isnull().any(axis=None)
+
+
+@pytest.mark.parametrize(
+    "cols,expected_cols", [(0, [0]), ([0, 1], [0, 1]), ([0, 2], [0]), (2, [])]
+)
+@pytest.mark.parametrize("types", ["URL", "EmailAddress", "NaturalLanguage"])
+def test_partial_dependence_not_allowed_types(types, cols, expected_cols, X_y_binary):
+    X, y = X_y_binary
```

Comment:
I think this is fine as is, but I wonder if we could make it more readable by either just using a X that is explicitly 3 cols or indicating that we're choosing col 2 because it's expected to be numeric? Just thinking that if we ever change X_y_binary to be not fully numeric, this could start changing the expected behavior of this test in unexpected ways.